dot/parachain/bitfield-distribution/bitfield_distribution_test.go

```diff
@@ -1215,3 +1215,242 @@ func TestBitfieldDistribution_ProcessIncomingPeerMessageEvent_Success(t *testing
 		t.Fatal("Test timed out")
 	}
 }
+
+func TestBitfieldDistribution_SendTrackedGossipMessage_noRelayParent(t *testing.T) {
+	overseerCh := make(chan any)
+
+	b := NewBitfieldDistribution(overseerCh)
+
+	dest := peer.ID("tester")
+
+	validatorIndex, validatorSet := prepareForValidators()
+
+	vec, err := parachaintypes.NewBitVec([]bool{true, false})
+	assert.Nil(t, err)
+
+	// perRelayParent data not exist for this hash
+	relayParent := common.Hash{1, 1, 1}
+	message := validationprotocol.CheckedBitfield{
+		Hash: relayParent,
+		CheckedSignedAvailabilityBitfield: parachaintypes.CheckedSignedAvailabilityBitfield{
+			Payload:        vec,
+			ValidatorIndex: validatorIndex,
+			Signature:      [64]byte{0},
+		},
+	}
+
+	go sendTrackedGossipMessage(b, dest, validatorSet[0], message, overseerCh)
+
+	select {
+	case <-overseerCh:
+		t.Fatal("should not receive any type of message")
+	case <-time.After(2 * time.Second):
+		t.Log("test complete")
+	}
```

Comment:
I don't have a great idea on how to test that something does not happen, but I'd really like to avoid waiting on timers in tests wherever possible. In general it can lead to flakiness (although not in this case) and in aggregate makes the entire test suite take longer and longer (bitfield distribution tests alone currently take over 8 seconds :-/).

How about something like this? IMHO we should always be waiting for any goroutines started in tests to terminate anyway.

```suggestion
        done := make(chan struct{})
                
	go func() {
	        sendTrackedGossipMessage(b, dest, validatorSet[0], message, overseerCh)
	        done <- struct{}
	}

	select {
	case <-overseerCh:
		t.Fatal("should not receive any type of message")
	case <-time.After(2 * time.Second):
		t.Fatal("test time out")		
	case <-done:
	}
```

Reply:
that is a great point, and I agree with you. I was also trying to figure out a better way for this case. One small thought on your suggestion, I think it would work if sending message to overseerCh inside the test target function(`sendTrackedGossipMessage()` in this case) is not running in another goroutine, otherwise `done` might be executed before `overseerCh` so that we can't be sure if something is sent via `overseerCh`

however In general I get your point, I guess let's impl things case by case for now.

Reply:
Hm, yeah good point about potential goroutines inside `sendTrackedGossipMessage()`. I can imagine someone adding one to remove a bottleneck or something and suddenly the test is flaky. Which is much worse than the test failing of course. :-/